lib/src/map/built_map.dart

```diff
@@ -30,20 +30,46 @@ abstract class BuiltMap<K, V> {
   ///
   /// Wrong: `new BuiltMap({1: '1', 2: '2', 3: '3'})`.
   ///
-  /// Right: `new BuiltMap<int, String>({1: '1', 2: '2', 3: '3'})`,
+  /// Right: `new BuiltMap<int, String>({1: '1', 2: '2', 3: '3'})`.
   ///
   /// Rejects nulls. Rejects keys and values of the wrong type.
   factory BuiltMap([map = const {}]) {
     if (map is _BuiltMap && map.hasExactKeyAndValueTypes(K, V)) {
       return map as BuiltMap<K, V>;
     } else if (map is Map || map is BuiltMap) {
-      return new _BuiltMap<K, V>.copyAndCheck(map.keys, (k) => map[k]);
+      return new _BuiltMap<K, V>.copyAndCheckTypes(map.keys, (k) => map[k]);
     } else {
       throw new ArgumentError(
           'expected Map or BuiltMap, got ${map.runtimeType}');
     }
   }
 
+  /// Instantiates with elements from a [Map].
+  ///
+  /// Must be called with a generic type parameter.
+  ///
+  /// Wrong: `new BuiltMap.from({1: '1', 2: '2', 3: '3'})`.
+  ///
+  /// Right: `new BuiltMap<int, String>.from({1: '1', 2: '2', 3: '3'})`.
+  ///
+  /// Rejects nulls. Rejects keys and values of the wrong type.
+  factory BuiltMap.from(Map map) {
+    return new _BuiltMap<K, V>.copyAndCheckTypes(map.keys, (k) => map[k]);
+  }
+
+  /// Instantiates with elements from a [Map<K, V>].
+  ///
+  /// `K` and `V` must not be `dynamic`.
+  ///
+  /// Wrong: `new BuiltMap.of({1: '1', 2: '2', 3: '3'})`.
+  ///
+  /// Right: `new BuiltMap<int, String>.of({1: '1', 2: '2', 3: '3'})`.
+  ///
+  /// Rejects nulls. Rejects keys and values of the wrong type.
+  factory BuiltMap.of(Map<K, V> map) {
+    return new _BuiltMap<K, V>.copyAndCheckForNull(map.keys, (k) => map[k]);
```

Comment:
Implementation wise, both are identical. I don't understand why.

Reply:
Not quite--`from` has to check the type and for nulls, `of` knows the type (because Dart 2 has types! woo!) but still has to check for nulls. So it calls `copyAndCheckForNull` instead of `copyAndCheckTypes`.

But the really _important_ difference is that the `of` method accepts a `Map<K, V>` instead of just a `Map`. This means type inference works :+1:

Reply:
Understood.